test/driver.cpp

```diff
@@ -15,123 +15,124 @@
 #include <fstream>
 #include <filesystem>
 #include <charconv>
-#include <array>
-#include <vector>
 #include <string>
 #include <string_view>
 #include <cstddef>
+#include <cassert>
 
 #include "tester.h"
 
 //must be defined in a unit-specific source file such as "array-test.cpp"
 void runTests();
 
-void processCmdLine(std::vector<std::string_view> arguments);
+void processCmdLine(char* arguments[], const std::size_t size);
+passReportMode getPassReportMode(const std::string_view& value, bool fileOutput);
+
+void replace_all(std::string& str, const std::string& substr, 
+   const std::string& new_substr);
 
 int main(int argc, char* argv[])
 {
-    std::vector<std::string_view> arguments;
-    for (int i = 0; i < argc; ++i)
-        arguments.emplace_back(argv[i]);
-    
-    processCmdLine(arguments);
+    processCmdLine(argv, argc);
 
    return getTestsFailed();
 }
 
-static const std::string defaultHeaderText("Running $exe");
-
-void processCmdLine(std::vector<std::string_view> arguments)
+void processCmdLine(char* arguments[], const std::size_t size)
 {
+   using std::string;
+   using std::string_view;
+   using std::size_t;
+   
+   //default values for cmd-line option names in name-value pairs
    bool printHeader{ true };
    bool printSummary{ true };
-   std::string headerText(defaultHeaderText);
-   std::string passReportMode("auto");
+   string headerText("Running $exe");
+   string_view passReportMode;
```

Comment:
As discussed [here](https://github.com/sigcpp/stl-lite/pull/13#discussion_r434258887), I suggest we initialize `passReportMode` to `"auto"` to make it more explicit that `auto` is the default pass report mode.
```suggestion
   string_view passReportMode("auto");
```

Reply:
Thanks @Ndemco for the suggestion. As I said earlier, I see your point, but I'm a bit worried that the literal "auto" now appears in two different places, far from each other. The way it is right now, only function `getPassReportMode` determines the exact spelling and logic of the default behavior. Implementing the suggestion changes that and introduces a point of failure in that if the default behavior and the text representing that is changed later, it needs to changed in two places and it is easy to forget changing one of them. What do you think?

Reply:
Yes, I see what you're saying @smurthys. I agree there should only be one place that the code needs to be changed if the default behavior / text is changed. Perhaps we keep `passReportMode` as an empty `string_view` and simply add a comment the states function `getPassReportMode` treats an empty `string_view` as the default `auto` value.